sdk/trace/simple_span_processor_test.go

```diff
@@ -150,6 +153,30 @@ func TestSimpleSpanProcessorShutdownOnEndConcurrency(t *testing.T) {
 	<-done
 }
 
+func TestSimpleSpanProcessorShutdownOnEndRace(t *testing.T) {
+	exporter := &testExporter{}
+	ssp := sdktrace.NewSimpleSpanProcessor(exporter)
+	tp := basicTracerProvider(t)
+	tp.RegisterSpanProcessor(ssp)
+
+	var wg sync.WaitGroup
+	wg.Add(2)
+
+	span := func(spanName string) {
+		defer wg.Done()
+		_, span := tp.Tracer("test").Start(context.Background(), spanName)
+		defer span.End()
```

Comment:
nit: not sure this needs to be defered

```suggestion
		span.End()
```